<p align="center">
<img src=https://github.com/Mathieu7483/Aiko78-Photgraphy/blob/main/img/python%20n%C3%A9eon%20carte%20%C3%A9l%C3%A9ctronique.png>
</p>

# 0x03. Python - Exceptions

# üìù Description

Ce projet explore la gestion des erreurs et des exceptions en Python. L'objectif est de comprendre la diff√©rence entre les deux et d'apprendre √† utiliser les blocs try, except, et finally pour cr√©er un code plus fiable et r√©silient. Les exercices abordent la gestion des erreurs de type (TypeError), des erreurs d'index (IndexError), des divisions par z√©ro (ZeroDivisionError), ainsi que la lev√©e d'exceptions personnalis√©es.

# üìÇ Contenu de l'exercice

Ce r√©pertoire contient les solutions des exercices du projet. Chaque fichier correspond √† une t√¢che sp√©cifique.

0-safe_print_list.py : Une fonction qui imprime en toute s√©curit√© un nombre x d'√©l√©ments d'une liste.

1-safe_print_integer.py : Une fonction qui imprime un entier de mani√®re s√©curis√©e, g√©rant les valeurs non-enti√®res.

2-safe_print_list_integers.py : Une fonction qui imprime uniquement les entiers d'une liste, en g√©rant les exceptions.

3-safe_print_division.py : Une fonction qui g√®re les divisions par z√©ro et s'assure d'imprimer un r√©sultat dans le bloc finally.

4-list_division.py : Une fonction qui divise deux listes √©l√©ment par √©l√©ment, en g√©rant plusieurs types d'erreurs (type, division par z√©ro, et index hors de port√©e).

5-raise_exception.py : Une fonction qui l√®ve une exception de type TypeError.

6-raise_exception_msg.py : Une fonction qui l√®ve une exception de type NameError avec un message personnalis√©.

7- 100-safe_print_integer_err.py : Cette fonction imprime un entier de mani√®re s√©curis√©e. Si la valeur n'est pas un entier, elle renvoie False et affiche le message d'erreur standard sur la sortie d'erreur (stderr), sans interrompre le programme.

8- 101-safe_function.py : Cette fonction ex√©cute une autre fonction de mani√®re s√©curis√©e. Si une exception se produit pendant l'ex√©cution, elle la g√®re, affiche l'erreur sur stderr et retourne None, permettant ainsi au programme de continuer son ex√©cution sans planter.

9- 102-magic_calculation.py: Cette t√¢che consiste √† reconstruire une fonction Python √† partir de son bytecode. L'objectif est de traduire les instructions de bas niveau en code Python fonctionnel, comprenant les boucles, les op√©rations et la gestion des exceptions.

# üõ†Ô∏è Pr√©requis

Environnement : Ubuntu 20.04 LTS

Interpr√©teur : Python 3.8.5

Ligne de code : Chaque fichier Python doit commencer par #!/usr/bin/python3

Formatage : Le code doit respecter les r√®gles de pycodestyle (version 2.7.*)

# üöÄ Compilation et ex√©cution

Pour ex√©cuter les scripts, assurez-vous que les fichiers sont ex√©cutables (chmod +x <nom_du_fichier.py>).

Exemple pour le fichier 0-main.py :

```Bash

./0-main.py
```

# ‚úçÔ∏è Auteur
[Mathieu GODALIER](https://github.com/Mathieu7483) - √âl√®ve en programmation √† la Holberton School

# ‚öñÔ∏è Licence

Ce projet est sous licence MIT. Pour plus de d√©tails, consultez le fichier LICENSE.

# üôè Remerciements

Ce projet a √©t√© r√©alis√© dans le cadre du cursus de programmation de l'√âcole Holberton. Un grand merci pour l'opportunit√© d'approfondir ma compr√©hension des exceptions en Python.